Tests/AsyncHTTPClientTests/HTTPConnectionPool+HTTP1StateTests.swift

```diff
@@ -102,13 +103,86 @@ class HTTPConnectionPool_HTTP1StateMachineTests: XCTestCase {
         XCTAssert(connections.isEmpty)
     }
 
+    func testCreatingAndFailingConnectionsWithoutRetry() {
+        struct SomeError: Error, Equatable {}
+        let elg = EmbeddedEventLoopGroup(loops: 4)
+        defer { XCTAssertNoThrow(try elg.syncShutdownGracefully()) }
+
+        var state = HTTPConnectionPool.StateMachine(
+            idGenerator: .init(),
+            maximumConcurrentHTTP1Connections: 8,
+            retryConnectionEstablishment: false
+        )
+
+        var connections = MockConnectionPool()
+        var queuer = MockRequestQueuer()
+
+        // for the first eight requests, the pool should try to create new connections.
+
+        for _ in 0..<8 {
+            let mockRequest = MockHTTPRequest(eventLoop: elg.next())
+            let request = HTTPConnectionPool.Request(mockRequest)
+            let action = state.executeRequest(request)
+            guard case .createConnection(let connectionID, let connectionEL) = action.connection else {
+                return XCTFail("Unexpected connection action")
+            }
+            XCTAssertEqual(.scheduleRequestTimeout(for: request, on: mockRequest.eventLoop), action.request)
+            XCTAssert(connectionEL === mockRequest.eventLoop)
+
+            XCTAssertNoThrow(try connections.createConnection(connectionID, on: connectionEL))
+            XCTAssertNoThrow(try queuer.queue(mockRequest, id: request.id))
+        }
+
+        // the next eight requests should only be queued.
+
+        for _ in 0..<8 {
+            let mockRequest = MockHTTPRequest(eventLoop: elg.next())
+            let request = HTTPConnectionPool.Request(mockRequest)
+            let action = state.executeRequest(request)
+            guard case .none = action.connection else {
+                return XCTFail("Unexpected connection action")
+            }
+            XCTAssertEqual(.scheduleRequestTimeout(for: request, on: mockRequest.eventLoop), action.request)
+            XCTAssertNoThrow(try queuer.queue(mockRequest, id: request.id))
+        }
+
+        // timeout all queued requests except for two
+
+        // fail all connection attempts
```

Comment:
I guess those can be removed.

Reply:
Good catch! Removed in https://github.com/swift-server/async-http-client/pull/625/commits/2cff156b95db93d9bc9bedb1b0c7890a827ad138